dot/parachain/backing/candidate_backing.go

```diff
@@ -5,124 +5,723 @@ package backing
+			}
+		case <-ctx.Done():
+			close(cb.SubSystemToOverseer)
+			close(chRelayParentAndCommand)
+		}
+	}
 }
 
 func (*CandidateBacking) Name() parachaintypes.SubSystemName {
 	return parachaintypes.CandidateBacking
 }
 
-func (cb *CandidateBacking) processMessages() {
-	for msg := range cb.OverseerToSubSystem {
-		// process these received messages by referencing
-		// https://github.com/paritytech/polkadot-sdk/blob/769bdd3ff33a291cbc70a800a3830638467e42a2/polkadot/node/core/backing/src/lib.rs#L741
-		switch msg.(type) {
-		case ActiveLeavesUpdate:
-			cb.handleActiveLeavesUpdate()
-		case GetBackedCandidates:
-			cb.handleGetBackedCandidates()
-		case CanSecond:
-			cb.handleCanSecond()
-		case Second:
-			cb.handleSecond()
-		case Statement:
-			cb.handleStatement()
-		default:
-			logger.Error("unknown message type")
-		}
+// processOverseerMessage processes incoming messages from overseer
+func (cb *CandidateBacking) processOverseerMessage(msg any, chRelayParentAndCommand chan RelayParentAndCommand) error {
+	// process these received messages by referencing
+	// https://github.com/paritytech/polkadot-sdk/blob/769bdd3ff33a291cbc70a800a3830638467e42a2/polkadot/node/core/backing/src/lib.rs#L741
+	switch msg := msg.(type) {
+	case ActiveLeavesUpdate:
+		cb.handleActiveLeavesUpdate()
+	case GetBackedCandidatesMessage:
+		cb.handleGetBackedCandidatesMessage()
+	case CanSecondMessage:
+		cb.handleCanSecondMessage()
+	case SecondMessage:
+		cb.handleSecondMessage()
+	case StatementMessage:
+		return cb.handleStatementMessage(msg.RelayParent, msg.SignedFullStatement, chRelayParentAndCommand)
+	default:
+		return errors.New("unknown message type")
 	}
+	return nil
 }
 
 func (cb *CandidateBacking) handleActiveLeavesUpdate() {
 	// TODO: Implement this #3503
 }
 
-func (cb *CandidateBacking) handleGetBackedCandidates() {
+func (cb *CandidateBacking) handleGetBackedCandidatesMessage() {
 	// TODO: Implement this #3504
 }
 
-func (cb *CandidateBacking) handleCanSecond() {
+func (cb *CandidateBacking) handleCanSecondMessage() {
 	// TODO: Implement this #3505
 }
 
-func (cb *CandidateBacking) handleSecond() {
+func (cb *CandidateBacking) handleSecondMessage() {
 	// TODO: Implement this #3506
 }
 
-func (cb *CandidateBacking) handleStatement() {
-	// TODO: Implement this #3507
+// Import the statement and kick off validation work if it is a part of our assignment.
+func (cb *CandidateBacking) handleStatementMessage(
+	relayParent common.Hash,
+	signedStatementWithPVD SignedFullStatementWithPVD,
+	chRelayParentAndCommand chan RelayParentAndCommand,
+) error {
+	rpState, ok := cb.perRelayParent[relayParent]
+	if !ok {
+		return fmt.Errorf("%w: %s", ErrStatementForUnknownRelayParent, relayParent)
+	}
+
+	summary, err := rpState.importStatement(cb.SubSystemToOverseer, signedStatementWithPVD, cb.perCandidate)
+	if err != nil {
+		return fmt.Errorf("importing statement: %w", err)
+	}
+
+	rpState.postImportStatement(cb.SubSystemToOverseer, summary)
```

Comment:
The function `postImportStatement` in their first line checks if the summary is nil and if so it calls `issueNewMisbehaviors` and here after calling we check the summary again and returns.

I would suggest
```
if summary == nil {
        logger.Debug("statement is nil")
	issueNewMisbehaviors(cb.SubSystemToOverseer, rpState.RelayParent, rpState.Table)
	return nil
}

rpState.postImportStatement(cb.SubSystemToOverseer, summary)
```

and then in the `postImportStatement` method remove the `sumary == nil` check

Reply:
The `postImportStatement` function will be called at multiple places.
This nil check is a behaviour of the `handleStatementMessage` function only.

	if summary == nil {
		logger.Debug("statement is nil")
		return nil
	}